src/index.spec.ts

```diff
@@ -290,283 +291,273 @@ describe('EppoJSClient E2E test', () => {
-        advanceTimers: true,
-        doNotFake: [
-          'Date',
-          'hrtime',
-          'nextTick',
-          'performance',
-          'queueMicrotask',
-          'requestAnimationFrame',
-          'cancelAnimationFrame',
-          'requestIdleCallback',
-          'cancelIdleCallback',
-          'setImmediate',
-          'clearImmediate',
-          'setInterval',
-          'clearInterval',
-        ],
+    global.fetch = jest.fn(() => {
+      const ufc = returnUfc(MOCK_UFC_RESPONSE_FILE);
+
+      return Promise.resolve({
+        ok: true,
+        status: 200,
+        json: () => Promise.resolve(ufc),
       });
-    });
+    }) as jest.Mock;
 
-    beforeEach(() => {
-      // We're creating a new instance for each test so we need to clear the underlying storage too
-      window.localStorage.clear();
-    });
+    mockLogger = td.object<IAssignmentLogger>();
 
-    afterEach(() => {
-      jest.clearAllTimers();
-      td.reset();
+    await init({
+      apiKey,
+      baseUrl,
+      assignmentLogger: mockLogger,
     });
 
-    afterAll(() => {
-      jest.useRealTimers();
+    jest.useFakeTimers({
+      advanceTimers: true,
+      doNotFake: [
+        'Date',
+        'hrtime',
+        'nextTick',
+        'performance',
+        'queueMicrotask',
+        'requestAnimationFrame',
+        'cancelAnimationFrame',
+        'requestIdleCallback',
+        'cancelIdleCallback',
+        'setImmediate',
+        'clearImmediate',
+        'setInterval',
+        'clearInterval',
+      ],
     });
 
-    it('default options', async () => {
-      td.replace(HttpClient.prototype, 'get');
-      let callCount = 0;
-      td.when(HttpClient.prototype.get(td.matchers.anything())).thenDo(() => {
-        if (++callCount === 1) {
-          // Throw an error for the first call
-          throw new Error('Intentional Thrown Error For Test');
-        } else {
-          // Return a mock object for subsequent calls
-          return mockConfigResponse;
-        }
-      });
+    // We're creating a new instance for each test so we need to clear the underlying storage too
+    window.localStorage.clear();
+  });
 
-      // By not awaiting (yet) only the first attempt should be fired off before test execution below resumes
-      const initPromise = init({
-        apiKey,
-        baseUrl,
-        assignmentLogger: mockLogger,
-      });
+  afterEach(() => {
+    td.reset();
+    jest.useRealTimers();
+    jest.clearAllTimers();
+  });
 
-      // Advance timers mid-init to allow retrying
-      await jest.advanceTimersByTimeAsync(maxRetryDelay);
+  it('default options', async () => {
+    let callCount = 0;
 
-      // Await so it can finish its initialization before this test proceeds
-      const client = await initPromise;
-      expect(callCount).toBe(2);
-      expect(client.getStringAssignment(flagKey, 'subject', {}, 'default-value')).toBe('control');
+    global.fetch = jest.fn(() => {
```

Comment:
converted all `td.replace(HttpClient` to `global.fetch` mocks

Reply:
💪